platform/security/encryption/platform-security-encryption-impl/src/main/java/ddf/security/encryption/impl/EncryptionServiceImpl.java

```diff
@@ -28,54 +42,75 @@
     synchronized (EncryptionServiceImpl.class) {
-      if (!new File(passwordDirectory.concat("/meta")).exists()) {
-        KeyczarTool.main(
-            new String[] {
-              "create", "--location=" + passwordDirectory, "--purpose=crypt", "--name=Password"
-            });
-        KeyczarTool.main(
-            new String[] {"addkey", "--location=" + passwordDirectory, "--status=primary"});
-      }
-      Crypter crypter = null;
+      File keysetFile = new File(keysetLocation);
+      InputStream keysetFileInputStream = null;
+      OutputStream keysetFileOutputStream = null;
       try {
-        crypter = new Crypter(passwordDirectory);
-      } catch (Exception e) {
-        LOGGER.debug(e.getMessage());
+        AeadConfig.register();
+        if (!keysetFile.exists()) {
+          keysetHandle = KeysetHandle.generateNew(AeadKeyTemplates.AES128_GCM);
+          keysetFileOutputStream = Files.newOutputStream(Paths.get(keysetLocation));
+          CleartextKeysetHandle.write(
+              keysetHandle, JsonKeysetWriter.withOutputStream(keysetFileOutputStream));
+        } else {
+          keysetFileInputStream = Files.newInputStream(Paths.get(keysetLocation));
+          keysetHandle =
+              CleartextKeysetHandle.read(JsonKeysetReader.withInputStream(keysetFileInputStream));
+        }
+        aead = AeadFactory.getPrimitive(keysetHandle);
+      } catch (GeneralSecurityException | IOException e) {
+        LOGGER.warn("Problem initializing Tink. Enable debug logging for more information.");
+        LOGGER.debug("", e);
+      } finally {
+        // close streams
+        try {
+          keysetFileInputStream.close();
+        } catch (IOException | NullPointerException ignore) {
+        }
+        try {
+          keysetFileOutputStream.close();
+        } catch (IOException | NullPointerException ignore) {
+        }
       }
-      this.crypter = crypter;
     }
   }
 
   /**
-   * Encrypts a plain text value using Keyczar.
+   * Encrypts a plain text value using Tink.
    *
    * @param plainTextValue The value to encrypt.
    */
   public synchronized String encrypt(String plainTextValue) {
     try {
-      return crypter.encrypt(plainTextValue);
-    } catch (Exception e) {
+      byte[] encryptedBytes = aead.encrypt(plainTextValue.getBytes(), ASSOCIATED_DATA);
+      return Base64.getEncoder().encodeToString(encryptedBytes);
+    } catch (GeneralSecurityException | NullPointerException e) {
       LOGGER.debug("Key and encryption service failed to set up. Failed to encrypt.", e);
       return plainTextValue;
     }
   }
 
   /**
-   * Decrypts a plain text value using Keyczar
+   * Decrypts a plain text value using Tink
    *
    * @param encryptedValue The value to decrypt.
    */
   public synchronized String decrypt(String encryptedValue) {
     try {
-      return crypter.decrypt(encryptedValue);
-    } catch (Exception e) {
+      byte[] encryptedBytes = Base64.getDecoder().decode(encryptedValue);
```

Comment:
📘 Can use byte arrays here without blowing up the heap because we are given a String.